src/types/UTF8OutPipeReader.cpp

```diff
@@ -32,13 +32,13 @@ UTF8OutPipeReader::UTF8OutPipeReader(wil::unique_hfile& outPipe) :
     dwRead += _dwPartialsLen;
     _dwPartialsLen = 0;
 
-    if (dwRead == 0) // quit if no data has been read and no cached data was left over
+    if (!fSuccess) // reading failed (we must check this first, because dwRead will also be 0.)
     {
-        return S_OK;
+        return E_FAIL;
     }
-    else if (!fSuccess) // reading failed
+    else if (dwRead == 0) // quit if no data has been read and no cached data was left over

```

Comment:
Once you update this, consider to make `else if` just an `if`. The `else` is completely meaningless here.